### PR TITLE
Update workflows and linters and fix linter warnings

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,27 +7,9 @@ on:
     branches:
     - "**"
   pull_request:
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (maraino) [inline .github/workflows/test.yml:11]: Why CI at 12am?

**Review by maraino — COMMENTED** [review]
looks good, but I don't get the schedule property.

File: .github/workflows/test.yml (hunk 1 above)
Comment (dopey) [reply .github/workflows/test.yml:11]: I want this to run once a day every single day. This way, if we upgrade the linter to v1.50.0 but don't update this repo in 6 months, we'll know that it's broken.

Comment (maraino) [reply .github/workflows/test.yml:11]: Sounds good.

**Pull request merged**